src/Arcus.Messaging.Pumps.ServiceBus.KeyRotation/ReAuthenticateOnRotatedCredentialsMessageHandler.cs

```diff
@@ -0,0 +1,82 @@
+        public ReAuthenticateOnRotatedCredentialsMessageHandler(string targetConnectionStringKey, AzureServiceBusMessagePump messagePump, ILogger<ReAuthenticateOnRotatedCredentialsMessageHandler> logger)
+        {
+            Guard.NotNullOrWhitespace(targetConnectionStringKey, nameof(targetConnectionStringKey), "Requires a non-blank secret key that points to the credentials that holds the connection string of the target message pump");
+            Guard.NotNull(messagePump, nameof(messagePump), $"Requires an message pump instance to restart when the message handler process an {nameof(SecretNewVersionCreated)} event");
+            Guard.NotNull(logger, nameof(logger), "Requires an logger instance to write diagnostic trace messages during the processing of the event");
+
+            _targetConnectionStringKey = targetConnectionStringKey;
+            _messagePump = messagePump;
+            _logger = logger;
+        }
+
+        /// <summary>
+        ///     Process a new message that was received
+        /// </summary>
+        /// <param name="message">Message that was received</param>
+        /// <param name="messageContext">Context providing more information concerning the processing</param>
+        /// <param name="correlationInfo">
+        ///     Information concerning correlation of telemetry and processes by using a variety of unique
+        ///     identifiers
+        /// </param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="message"/> is <c>null</c>.</exception>
+        /// <exception cref="CloudException">Thrown when the <paramref name="message"/> doesn't represent an <see cref="SecretNewVersionCreated"/> event.</exception>
+        public async Task ProcessMessageAsync(
+            CloudEvent message,
+            AzureServiceBusMessageContext messageContext,
+            MessageCorrelationInfo correlationInfo,
+            CancellationToken cancellationToken)
+        {
+            Guard.NotNull(message, nameof(message), "Cannot invalidate Azure KeyVault secret from a 'null' CloudEvent");
+
+            _logger.LogTrace("Receiving new Azure Key Vault notification...");
+            
+            var secretNewVersionCreated = message.GetPayload<SecretNewVersionCreated>();
+            if (secretNewVersionCreated is null)
+            {
+                throw new CloudException(
+                    "Azure Key Vault job cannot map Event Grid event to CloudEvent because the event data isn't recognized as a 'SecretNewVersionCreated' schema");
+            }
+
+            if (_targetConnectionStringKey == secretNewVersionCreated.ObjectName)
+            {
+                _logger.LogInformation("Received Azure Key vault 'Secret New Version Created' event, restarting target message pump {JobId}", _messagePump.JobId);
```

Comment:
This is maybe a verbose one which I would combine with an event, personally

Reply:
Not sure what you mean 😅 .

Reply:
I would change this to Trace, but maybe call LogEvent afterwards as well ~ "Message Pump Restarted" or so?

Reply:
Yes, did that! Resolving!